src/encoded/static/components/search.js

```diff
@@ -711,30 +711,56 @@ TypeTerm.propTypes = {
     total: PropTypes.number.isRequired,
 };
 
+// Sanitize user input and facet terms for comparison: convert to lowercase, remove white space and asterisks (which cause regular expression error)
+const sanitizedString = inputString => inputString.toLowerCase().replace(/ /g, '').replace(/[*]/g, '');
 
 class Facet extends React.Component {
     constructor() {
         super();
 
         // Set initial React commponent state.
         this.state = {
-            facetOpen: false,
+            filteredTerms: null,
+            initialState: true,
         };
 
         // Bind `this` to non-React methods.
-        this.handleClick = this.handleClick.bind(this);
+        this.handleSearch = this.handleSearch.bind(this);
     }
 
-    handleClick() {
-        this.setState(prevState => ({ facetOpen: !prevState.facetOpen }));
+    componentDidMount() {
+        this.setState({ initialState: false });
+    }
+
+    handleSearch(event) {
+        // search term entered by the user
+        const filterVal = String(sanitizedString(event.target.value));
+
+        // which facet terms match the search term entered by the user
+        let terms = this.props.facet.terms.filter((term) => {
+            if (term.doc_count > 0) {
+                const termKey = sanitizedString(term.key);
+                if (termKey.match(filterVal)) {
+                    return term;
+                }
+                return false;
+            }
+            return false;
+        });
```

Comment:
Since this works (and since I saw it before and didn’t say anything) I would consider this feedback optional. `handleSearch` is called with every keystroke and what it essentially does is see if event.target.value is in the terms. `this.props.facet.terms` doesn’t change unless the page re-renders, so I’m thinking maybe the filtered, sanitized terms should be cached in the constructor, and then handleSearch just needs to see if the sanitized `event.target.value` is in this cache or not.